apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -21,14 +26,18 @@ export class CardComponent {
 
   constructor(
     private teacherStore: TeacherStore,
-    private studentStore: StudentStore
+    private studentStore: StudentStore,
+    private cityStore: CityStore
   ) {}
 
   addNewItem() {
+    console.log(this.list);
     if (this.type === CardType.TEACHER) {
       this.teacherStore.addOne(randTeacher());
     } else if (this.type === CardType.STUDENT) {
       this.studentStore.addOne(randStudent());
+    } else if (this.type === CardType.CITY) {
+      this.cityStore.addOne(randomCity());
```

Comment:
This component should not have any logic inside. All the logic should be handle by the parent. Otherwise, everytime you need to add a new card, you will have to change this component. Not ideal. 

Look at ngTemplateOutlet 😉